tapdb/sqlc/migrations/000036_asset_commit.up.sql

```diff
@@ -0,0 +1,133 @@
+    output_index INTEGER,
+
+    -- The internal key used for the commitment output.
+    internal_key_id BIGINT NOT NULL REFERENCES internal_keys(key_id),
+
+    -- The taproot output key used for the commitment output.
+    output_key BLOB NOT NULL CHECK(length(output_key) = 33),
+
+    -- The block header of the block mining the commitment transaction.
+    block_header BLOB,
+
+    -- The block height at which the commitment transaction was confirmed.
+    -- Can be NULL if the transaction is not yet confirmed.
+    block_height INTEGER,
+
+    -- The merkle proof demonstrating the commitment's inclusion in the block.
+    merkle_proof BLOB,
+
+    -- The root hash of the supply commitment at this snapshot.
+    supply_root_hash BLOB,
+
+    -- The root sum of the supply commitment at this snapshot.
+    supply_root_sum BIGINT
+);
+
+-- Main table tracking the state machine instance per asset group.
+CREATE TABLE supply_commit_state_machines (
+    -- The tweaked group key identifying the asset group's state machine.
+    group_key BLOB PRIMARY KEY CHECK(length(group_key) = 33),
+
+    -- The current state of the state machine.
+    current_state_id INTEGER NOT NULL REFERENCES supply_commit_states(id),
+
+    -- The latest successfully committed supply state on chain.
+    -- Can be NULL if no commitment has been made yet.
+    latest_commitment_id BIGINT REFERENCES supply_commitments(commit_id)
+);
+
+-- Table tracking a pending state transition for a state machine.
+CREATE TABLE supply_commit_transitions (
+    transition_id INTEGER PRIMARY KEY,
+
+    -- Reference back to the state machine this transition belongs to.
+    state_machine_group_key BLOB NOT NULL REFERENCES supply_commit_state_machines(group_key),
+
+    -- The commitment being replaced by this transition.
+    -- Can be NULL if this is the first commitment.
+    old_commitment_id BIGINT REFERENCES supply_commitments(commit_id),
+
+    -- The new commitment that this transition aims to create.
+    -- Can be NULL initially, before the commitment details are created.
+    new_commitment_id BIGINT REFERENCES supply_commitments(commit_id),
+
+    -- The chain transaction that, once confirmed, will finalize this transition.
+    -- Can be NULL until the transaction is created and signed.
+    pending_commit_txn_id BIGINT REFERENCES chain_txns(txn_id),
+
+    -- Indicates if this transition has been successfully completed and committed.
+    finalized BOOLEAN NOT NULL DEFAULT FALSE,
+
+    -- Timestamp when this transition was initiated.
+    creation_time TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP
+);
+
+-- Table storing individual update events associated with a pending transition.
+CREATE TABLE supply_update_events (
+    event_id INTEGER PRIMARY KEY,
+
+    -- Reference to the state transition this event is part of.
+    transition_id BIGINT NOT NULL REFERENCES supply_commit_transitions(transition_id) ON DELETE CASCADE,
```

Comment:
an update event can't be persisted unless it references an existing transition, is that correct?
If so, then if we're waiting for tx conf then new update events can't be persisted?

If true, then I don't think it matters a great deal, but from a UX perspective, we will need to return an error to the user along the lines of: unconfirmed supply tx, can't accept new asset ignore right now.